Makefile

```diff
@@ -198,98 +198,116 @@ tofnd-client:
-# For some reason ibc expects confio proto files to be in the main folder
-CONFIO_TYPES        	= third_party/proto
-
-proto-update-deps:
-	@echo "Updating Protobuf deps"
-	@mkdir -p $(GOGO_PROTO_TYPES)
-	@curl -sSL $(GOGO_PROTO_URL)/gogoproto/gogo.proto > $(GOGO_PROTO_TYPES)/gogo.proto
-
-	@mkdir -p $(GOOGLE_API_TYPES)
-	@curl -sSL $(GOOGLE_API_URL)/annotations.proto > $(GOOGLE_API_TYPES)/annotations.proto
-	@curl -sSL $(GOOGLE_API_URL)/http.proto > $(GOOGLE_API_TYPES)/http.proto
-
-	@mkdir -p $(COSMOS_PROTO_TYPES)
-	@curl -sSL $(COSMOS_PROTO_URL)/cosmos.proto > $(COSMOS_PROTO_TYPES)/cosmos.proto
-
-## Importing of tendermint protobuf definitions currently requires the
-## use of `sed` in order to build properly with cosmos-sdk's proto file layout
-## (which is the standard Buf.build FILE_LAYOUT)
-## Issue link: https://github.com/tendermint/tendermint/issues/5021
-	@mkdir -p $(TM_ABCI_TYPES)
-	@curl -sSL $(TM_URL)/abci/types.proto > $(TM_ABCI_TYPES)/types.proto
-
-	@mkdir -p $(TM_VERSION)
-	@curl -sSL $(TM_URL)/version/types.proto > $(TM_VERSION)/types.proto
-
-	@mkdir -p $(TM_TYPES)
-	@curl -sSL $(TM_URL)/types/types.proto > $(TM_TYPES)/types.proto
-	@curl -sSL $(TM_URL)/types/evidence.proto > $(TM_TYPES)/evidence.proto
-	@curl -sSL $(TM_URL)/types/params.proto > $(TM_TYPES)/params.proto
-	@curl -sSL $(TM_URL)/types/validator.proto > $(TM_TYPES)/validator.proto
-	@curl -sSL $(TM_URL)/types/block.proto > $(TM_TYPES)/block.proto
-
-	@mkdir -p $(TM_CRYPTO_TYPES)
-	@curl -sSL $(TM_URL)/crypto/proof.proto > $(TM_CRYPTO_TYPES)/proof.proto
-	@curl -sSL $(TM_URL)/crypto/keys.proto > $(TM_CRYPTO_TYPES)/keys.proto
-
-	@mkdir -p $(TM_LIBS)
-	@curl -sSL $(TM_URL)/libs/bits/types.proto > $(TM_LIBS)/types.proto
-
-	@mkdir -p $(TM_P2P)
-	@curl -sSL $(TM_URL)/p2p/types.proto > $(TM_P2P)/types.proto
-
-	@mkdir -p $(CONFIO_TYPES)
-	@curl -sSL $(CONFIO_URL)/proofs.proto > $(CONFIO_TYPES)/proofs.proto
-## insert go package option into proofs.proto file
-## Issue link: https://github.com/confio/ics23/issues/32
-	@./scripts/sed.sh $(CONFIO_TYPES)/proofs.proto
-
-	@./scripts/proto-copy-cosmos-sdk.sh
-
-.PHONY: proto-all proto-gen proto-gen-any proto-format proto-lint proto-check-breaking proto-update-deps
+	@$(protoImage) buf breaking --against $(HTTPS_GIT)#branch=main
+
+###############################################################################
+#                             Swagger Configuration                           #
+###############################################################################
+
+SWAGGER_DIR=./swagger-proto
+THIRD_PARTY_DIR=$(SWAGGER_DIR)/third_party
+
+# Dependency versions
+# COSMOS_SDK_VERSION=$(go list -m -json github.com/cosmos/cosmos-sdk | jq -r .Version)
+# IBC_VERSION=$(go list -m all | grep "github.com/cosmos/ibc-go" | cut -d' ' -f2)
+# WASMD_VERSION=$(go list -m -json github.com/CosmWasm/wasmd |  jq -r .Version)
+COSMOS_SDK_VERSION=v0.47.15
+IBC_VERSION=v7.8.0
+WASMD_VERSION=v0.46.0
+
+proto-swagger-download-deps:
```

Comment:
does this mean we can never cache proto dependencies? it seems like it's cleaning everything out whenever this is called

Reply:
Yes, proto cache is removed. It's now managed by buf build.
Swagger gen needs access to proto files; it downloads those on the fly and cleans up after building.